finagle-core/src/test/scala/com/twitter/finagle/tracing/DefaultTracingTest.scala

```diff
@@ -40,8 +40,10 @@ class DefaultTracingTest extends FunSuite with StringClient with StringServer {
   /**
    * Ensure all annotations have the same TraceId (unique to server and client though)
    * Ensure core annotations are present and properly ordered
+   * NOTE by timxzl: f must also return a finalizer: () => Unit that properly closes the services built by f
+   * otherwise the tests here will prevent ExitGuard from exiting and interfere with ExitGuardTest
    */
-  def testCoreTraces(f: (Tracer, Tracer) => (Service[String, String])) {
+  def testCoreTraces(f: (Tracer, Tracer) => (Service[String, String], () => Unit)) {
```

Comment:
feel free to ignore: I think this would be cleaner to return a Closable instead of () => Unit, which more explicitly points out that there's a resource that needs closing, rather than that some mutation needs to be done, which is more generic.